File: WorldFeed/Client/src/app/core/Public/Parameters/AccountClient/GetUsersRequestingFriendshipParameters.ts
import {TwitterLimits} from "../../Settings/TwitterLimits";
import {GetCursorUsersOptionalParameters, IGetCursorUsersOptionalParameters} from "../Optionals/GetCursorUsersOptionalParameters";
import {IGetUserIdsRequestingFriendshipParameters} from "./GetUserIdsRequestingFriendshipParameters";

// For more information visit : https://developer.twitter.com/en/docs/accounts-and-users/follow-search-get-users/api-reference/get-friendships-incoming
export interface IGetUsersRequestingFriendshipParameters extends IGetCursorUsersOptionalParameters, IGetUserIdsRequestingFriendshipParameters {
  // Page size when retrieving the users objects from Twitter
  getUsersPageSize: number;
}

export class GetUsersRequestingFriendshipParameters extends GetCursorUsersOptionalParameters implements IGetUsersRequestingFriendshipParameters {
  constructor(source?: IGetUsersRequestingFriendshipParameters) {
    if (source) {
      super(source);
      this.getUsersPageSize = source.getUsersPageSize;
      return;
    }

    this.getUsersPageSize = TwitterLimits.DEFAULTS.USERS_GET_USERS_MAX_SIZE;

  }

  public getUsersPageSize: number;
}


//   public GetUsersRequestingFriendshipParameters()
//   {
//     GetUsersPageSize = TwitterLimits.DEFAULTS.USERS_GET_USERS_MAX_SIZE;
//   }
//
//   public GetUsersRequestingFriendshipParameters(IGetUsersRequestingFriendshipParameters source) : base(source)
// {
//   if (source == null)
//   {
//     GetUsersPageSize = TwitterLimits.DEFAULTS.USERS_GET_USERS_MAX_SIZE;
//     return;
//   }
//
//   GetUsersPageSize = source.GetUsersPageSize;
// }

File: Sharebook/Server/src/Services/Profile/Sharebook.Profile.API/Application/Parameters/AccountClient/GetUsersRequestingFriendshipParameters.cs
namespace Sharebook.Profile.Application.Parameters.AccountClient
{
    using Sharebook.Common.Public.Parameters.Optionals;
    using Sharebook.Common.Settings;

    /// <summary>
    /// For more information visit : https://developer.twitter.com/en/docs/accounts-and-users/follow-search-get-users/api-reference/get-friendships-incoming
    /// </summary>
    /// <inheritdoc />
    public interface IGetUsersRequestingFriendshipParameters : IGetCursorUsersOptionalParameters, IGetUserIdsRequestingFriendshipParameters
    {
        /// <summary>
        /// Page size when retrieving the users objects from Twitter
        /// </summary>
        int GetUsersPageSize { get; set; }
    }

    /// <inheritdoc />
    public class GetUsersRequestingFriendshipParameters : GetCursorUsersOptionalParameters, IGetUsersRequestingFriendshipParameters
    {
        public GetUsersRequestingFriendshipParameters()
        {
            GetUsersPageSize = WorldFeedLimits.DEFAULTS.USERS_GET_USERS_MAX_SIZE;
        }

        public GetUsersRequestingFriendshipParameters(IGetUsersRequestingFriendshipParameters source) : base(source)
        {
            if (source == null)
            {
                GetUsersPageSize = WorldFeedLimits.DEFAULTS.USERS_GET_USERS_MAX_SIZE;
                return;
            }

            GetUsersPageSize = source.GetUsersPageSize;
        }

        /// <inheritdoc/>
        public int GetUsersPageSize { get; set; }
    }
}
